[WIP] Simplify 'Add a theme' form

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies add a theme forms
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  |  Theme & Logo > Theme. Everything should work exactly the same

:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType by ImportThemeType. This means if any module extends any of this type they will get an exception upon upgrading to PS version containing changes in this PR.
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21243)
<!-- Reviewable:end -->

zuk3975 committed:
rollback removing currency search action. Deprecate instead

zuk3975 committed:
rollback removing taxes search action. Deprecate instead

zuk3975 committed:
apply cs-fixer

Commit:
theme: add caching on Yaml parsing, by using Prestashop's YamlParser instead of direct Symfony's one

matks committed:
Fix codestyle and code review feedbacks

JevgenijVisockij committed:
Simplified theme form

JevgenijVisockij committed:
Adding back accidentally deleted lines

JevgenijVisockij committed:
Removed uneeded space

matks commented on src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig:
```diff
@@ -101,6 +101,12 @@
   {{ block('form_help') }}
 {% endblock switch_widget %}
 
+{%- block url_widget -%}
```
This modification is in `prestashop_ui_kit.html.twig` and not `prestashop_ui_kit_base.html.twig` is that intended ?

JevgenijVisockij replied:
Good catch, yeah I think it should be in base.

Progi1984 committed:
Fixed Rebase

Progi1984 committed:
Harmonize controller grid search action (Logs, SQLManager, WebService, Meta)

Progi1984 committed:
Fixed Code Reviews

JevgenijVisockij committed:
Moved url widget to ui_kit_base

JevgenijVisockij commented:
Blocked until https://github.com/PrestaShop/PrestaShop/pull/21878 is merged since it also deals with problem in file type.

matks committed:
Move Product pack classes in Pack subdomain in adapter

Commit:
Fix buttons on order view page

davidglezz committed:
Improve TypedRegexValidator->getPattern()

davidglezz committed:
Use constants instead of string literals

davidglezz committed:
Avoid code/logic duplication

Commit:
Swap default theme sass files to scss

Commit:
Last files

nesrineabdmouleh committed:
Add create order tests

nesrineabdmouleh committed:
Add go to view order page tests

matks committed:
Merge pull request #22588 from Progi1984/issue21200

Added cron for dumping JS Routing

Commit:
Separate gifts from products in Order cart presenter

Commit:
Switch last files

Commit:
Switch combined property into multiple because of SCSS

kpodemski committed:
bring back vat number to order view

kpodemski committed:
action

kpodemski committed:
skip job if there's a [merge] in pr title

kpodemski committed:
test title

kpodemski committed:
fix action

kpodemski committed:
cleaning

kpodemski committed:
avoid BC

kpodemski committed:
missing phpdocs

Progi1984 committed:
Fixed body in Cron Update JS Routing

kpodemski committed:
add empty line

Co-authored-by: Mathieu Ferment <mathieu.ferment@prestashop.com>

PierreRambaud committed:
Merge pull request #22845 from Progi1984/issue21200

Fixed body in Cron Update JS Routing

matks committed:
Merge pull request #22820 from Progi1984/gaUpdateSetupPhp

Update Setup PHP Actions to v2

nesrineabdmouleh committed:
CHange some ids and add verification tests

Progi1984 committed:
Merge pull request #22808 from PierreRambaud/update/do-not-log

Do not log addons requests urls, we don't want it in logs

jolelievre committed:
Revert service that must not move

jolelievre committed:
Fix and separate pack services definitions

jolelievre committed:
Move all core pack classes and interfaces into dedicated Pack namespace

jolelievre committed:
Final services renaming

jolelievre committed:
Move customization handlers into proper namespace

JevgenijVisockij committed:
Add validation for correct image type to add and edit language

jolelievre committed:
Move customization services into proper namespace

JevgenijVisockij committed:
Added proper errors for image validation

JevgenijVisockij committed:
Fixed typos in language deletion

Progi1984 committed:
Merge pull request #22736 from NeOMakinG/issue21832

Update the UIKit to the latest release

kpodemski committed:
Merge pull request #22849 from jolelievre/move-customization-namespace

Move customization into product subdomain namespace

kpodemski committed:
Merge pull request #22555 from matks/move-subdomain-adapter-pack

Move Product pack classes in Pack subdomain in adapter

nesrineabdmouleh committed:
Add the full scenario

JevgenijVisockij committed:
Added proper message for failed main language deletion

nesrineabdmouleh committed:
Fix test context

JevgenijVisockij committed:
CS fixer fix

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Delete bad line on categories tests

Commit:
Fix icons before double dot not valid using SCSS

PierreRambaud committed:
Merge pull request #22842 from kpodemski/fix/21294

Block PR if it has merge commits

Commit:
Add test 'View vouchers on FO account Page'

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Add some fixes

Commit:
Fix base context

matks committed:
Updated JS Routing file

Progi1984 committed:
Merge pull request #22861 from PrestaShop/create-pull-request/patch

Updated JS Routing file

Commit:
Add test 'Check number of orders in order history page'

kpodemski committed:
fix reference in shopping carts view

nesrineabdmouleh committed:
Add scenario

nesrineabdmouleh committed:
Add methods and selectors

nesrineabdmouleh committed:
Rename test file

nesrineabdmouleh committed:
Add test create shop

nesrineabdmouleh committed:
Add edit tests

nesrineabdmouleh committed:
Complete the scenario

Prestaworks committed:
Update Carrier.php

Fixing issue 22877 and making the code a bit more easy to read.

Prestaworks committed:
Update Carrier.php

davidglezz committed:
Fix style

Commit:
Change create order buttons position and wording

SimonGrn committed:
Merge pull request #22876 from nesrineabdmouleh/FilterSortPaginationLogs

Add test 'Filter,sort and pagination logs'

SimonGrn committed:
Merge pull request #22879 from nesrineabdmouleh/functionalCRUDshopGroup

Add test 'CRUD shop group'

SimonGrn committed:
Merge pull request #22762 from nesrineabdmouleh/functionalFilterSortPaginationShopGroups

Add test 'Filter sort pagination shop groups'

nesrineabdmouleh committed:
First commit

PierreRambaud committed:
Update src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig

PierreRambaud committed:
Update src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/customer.html.twig

Commit:
Fix shopping cart tests for the nightly

nesrineabdmouleh committed:
Add test for update tax excluded

Progi1984 committed:
Merge pull request #19818 from Progi1984/issue12168

Added Countries Translations (Installation & LocalizationPack)

Prestaworks committed:
Update Carrier.php

Prestaworks committed:
Delete Carrier.php

Prestaworks committed:
Create Carrier.php

Commit:
Update text

Prestaworks committed:
Update Carrier.php

Commit:
Adjust items alignment of order view header items

jolelievre committed:
Merge pull request #18136 from zuk3975/fix/common-search-dev

Use grid common search action where possible (1.7.8 target)

Progi1984 committed:
Merge pull request #22890 from boubkerbribri/Fix-shopping-carts-tests

Fix shopping cart tests for the nightly

Progi1984 committed:
Merge pull request #22874 from boubkerbribri/account-order-history

Add test 'Check number of orders in order history page'

Progi1984 committed:
Merge pull request #22855 from boubkerbribri/user-discount-fo

Add test 'View vouchers on FO account Page'

Progi1984 committed:
Merge pull request #22856 from nesrineabdmouleh/functionalCheckCustomerBlockInOrderPage

Add test 'Check customer block' in order page

Commit:
Fix on very small mobile

Commit:
Add dropdown with more actions

nesrineabdmouleh committed:
Add ids in shipping>preferences page

nesrineabdmouleh committed:
Add all scenario

matks committed:
Merge pull request #22805 from Progi1984/issue22625

BO - Orders page - Cannot delete a product restricted by a cart rule

nesrineabdmouleh committed:
Fix test identifiers

nesrineabdmouleh committed:
Fix test title

Progi1984 committed:
BO - Create Order - Updated product list when adresses changed

Commit:
Fix customer block test

Commit:
Fix user discount test

Commit:
Fix step identifiers

Progi1984 committed:
Merge pull request #22863 from kpodemski/fix/22854

Fix reference in Shopping carts view

Progi1984 committed:
Merge pull request #22900 from nesrineabdmouleh/functionalHandlingCharges

Add test 'Test handling charges for carriers in FO'

Progi1984 committed:
Merge pull request #22895 from NeOMakinG/issue22891

Adjust items alignment of order view header

Progi1984 committed:
Merge pull request #22912 from boubkerbribri/fix-nightly-20-01-2021

Fix nightly tests failing on 20-01-2021

Progi1984 committed:
Merge pull request #22909 from Progi1984/issue22243

BO - Create Order - Updated product list when adresses changed

Commit:
Fix lint

jolelievre committed:
Improve TextWithLengthCounterType to add attributes on embedded input

jolelievre committed:
Create and integrate SEOType into ProductType, also add preview button

jolelievre committed:
settings

jolelievre committed:
Add TextWithLengthCounter to the available components list

jolelievre committed:
Integrate SEO tab and form into product page

jolelievre committed:
Integrate SEO js feature like Serp preview and TextWithLengthCounter

jolelievre committed:
Fix friendly url placeholder in Serp component

jolelievre committed:
Integrate url provider in ProductType

jolelievre committed:
Add class for margin on each tab

jolelievre committed:
Integrate SEOCommandBuilder with tests

jolelievre committed:
Update product data provider with SEO options

jolelievre committed:
Check for changes on keyup

atomiix committed:
Merge pull request #22884 from Prestaworks/Fix-for-issue-22877

Fix carrier depth, height and width limits

nesrineabdmouleh committed:
Add test 'Add shop URL'

kpodemski committed:
new hooks on grid component table

kpodemski committed:
move hooks outside form

nesrineabdmouleh committed:
Add test 'Delete shop URL'

nesrineabdmouleh committed:
Fix test identifiers

atomiix committed:
Merge pull request #20446 from davidglezz/fix-cache-class

Fix cache class

PululuK committed:
Merge pull request #22922 from kpodemski/feature/22848

New hooks on grid component table

Commit:
Fix nightly 21-01-2021 on 1.7.7.x

boubkerbribri committed:
Update tests/UI/campaigns/functional/FO/03_userAccount/03_userDiscounts.js

Co-authored-by: Progi1984 <progi1984@gmail.com>

Progi1984 committed:
Merge pull request #22927 from boubkerbribri/fix-nightly-21-01-2021

Fix nightly 21-01-2021 on 1.7.7.x

matks committed:
Make better message for default language deletion

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

PierreRambaud committed:
Merge pull request #19231 from Kioob/fix-theme-yamlparser

Add caching on Theme Yaml parsing

matks committed:
Update GitHub Action actions/checkout to v2

nesrineabdmouleh committed:
Add scenario

Commit:
Fix undefined index when iso doesn't exists in language list

PierreRambaud committed:
Merge pull request #22887 from davidglezz/improve-TypedRegexValidator-getPattern

Improve TypedRegexValidator->getPattern()

Commit:
Fix product list hover on catalog mode

atomiix committed:
Function getCartPriceFromCatalog can return null

PierreRambaud committed:
Merge pull request #22935 from matks/update-ga-checkout-v2

Update GitHub Action actions/checkout to v2

nesrineabdmouleh committed:
Add sort tests

PierreRambaud committed:
Merge pull request #22841 from kpodemski/fix/22830

Bring back vat number to address on order view

nesrineabdmouleh committed:
Add sort tests

jolelievre committed:
Merge pull request #22944 from PierreRambaud/fix/installer

Fix undefined index when iso doesn't exists in language list

itsvahid committed:
Fix product image form display in mobile view

Increase in height of caption textarea in product images form exceeded the parent div height.
Product images form did not show up in mobile view after clicking on the previews to edit.

mvorisek committed:
No unlimited checkout depth

mvorisek committed:
Delete parameters.yml.travis

mvorisek committed:
Fix config path

Progi1984 committed:
Merge pull request #22768 from PierreRambaud/fix/22615

Exclude gifts from getOrderTotal when selected ONLY_PRODUCTS

Progi1984 committed:
Merge pull request #22967 from mvorisek/patch-1

Delete parameters.yml.travis

Commit:
Add on new theme

Progi1984 committed:
Merge pull request #22925 from nesrineabdmouleh/updateCRUDShopTest

Add test 'Add and delete shop URL'

Progi1984 committed:
Merge pull request #22954 from nesrineabdmouleh/functionalSortShopGroups

Add test 'Sort shop groups'

Commit:
Run on tests

Commit:
Run on tests campaigns

Commit:
Run on themes

Progi1984 committed:
Merge pull request #22662 from NeOMakinG/issue22660

Fix buttons height on mobile on order view page

Progi1984 committed:
Merge pull request #22955 from itsvahid/develop

Fix product image form display in mobile view

Commit:
Add hook

matks committed:
Merge pull request #21878 from JevgenijVisockij/feature/simplify-edit-language

Simplified add language form

zuk3975 committed:
Introduce Quantity type with choice providers

Progi1984 committed:
Merge pull request #22947 from atomiix/fix/22812

Fix Shopping Cart page in BO throwing an exception

Progi1984 committed:
Bump version to 1.7.7.2

nesrineabdmouleh committed:
Add all scenario

zuk3975 committed:
create command builder for quantity tab

nesrineabdmouleh committed:
Add some fixes

nesrineabdmouleh committed:
Fix test identifiers

zuk3975 committed:
add form constraints

Commit:
Remove header button on mobile and push these into new floating component

nesrineabdmouleh committed:
Add test description

zuk3975 committed:
Form data provider. Quantity part

zuk3975 committed:
enable "save" button on datepicker changeevent (dp.change)

PierreRambaud committed:
Merge pull request #22966 from mvorisek/faster_checkout

Faster checkout - depth of 0 means unlimited

Commit:
Hide shop name on mobile

zuk3975 committed:
add test for command builder

zuk3975 committed:
fix available_date null pointer

Progi1984 committed:
Merge pull request #22985 from nesrineabdmouleh/functionalSetRequiredFieldsAddress

Add test 'Set required fields for address page'

zuk3975 committed:
OutOfStockTypeChoiceProvider test

zuk3975 committed:
PackStockTypeChoiceProviderTest

zuk3975 committed:
add abstract ChoiceProviderTestCase

zuk3975 committed:
add formType in template and tabs

zuk3975 committed:
rename Quantity -> Stock occurrences

Commit:
Align labels to right and space on red star

Progi1984 committed:
Merge pull request #22417 from tswfi/breadcrumbs_rich_data_home_page

Add breadcrumb rich data on the frontpage

Progi1984 committed:
Merge pull request #22945 from NeOMakinG/issue22715

Fix product list hover and overflow on catalog mode

atomiix committed:
Merge pull request #22982 from Progi1984/bumpVersion1772

Bump version to 1.7.7.2

Commit:
Fix compare function of dashboard calendar

Progi1984 committed:
Merge pull request #22986 from zuk3975/m/prod-form/qty

Integrate product quantities form and related command

Commit:
Update ps config eslint

Commit:
Remove unknown todo

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

Commit:
Remove unknown todo

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

Commit:
Change is to prop

matks committed:
Update Symfony debug toolbar link to developer documentation

Seb33300 committed:
Fix product search by ISBN

Commit:
Fix alerts

Commit:
Fix ship alerts

atomiix committed:
Merge pull request #22975 from NeOMakinG/add-breakline-rule

Add breakline rule of ESLint

Progi1984 committed:
Merge pull request #22988 from NeOMakinG/issue22987

Hide shop name on mobile

Commit:
Rebase

Progi1984 committed:
Merge pull request #22997 from NeOMakinG/issue-22973

Fix compare function of dashboard's calendar

Commit:
Change display

matks committed:
Merge pull request #22996 from NeOMakinG/issue22873

Align labels to right and space on red star

Progi1984 committed:
Merge pull request #23003 from matks/matks-patch-1

Update Symfony debug toolbar link to developer documentation

Progi1984 committed:
Merge pull request #22984 from NeOMakinG/issue22978

Remove header button on mobile and push these into new floating btn

Progi1984 committed:
Merge pull request #22953 from nesrineabdmouleh/functionalFilterSortAndPaginationShops

Add test 'Filter, sort and paginaion shops'

Commit:
Fix feedbacks from QA

Commit:
Review feedback

Progi1984 committed:
Merge pull request #22813 from NeOMakinG/issue22806

Remove box-shadows from classic theme

matthieu-rolland committed:
Merge pull request #22772 from NeOMakinG/issue22763

Swap default theme sass files to scss

jolelievre committed:
Little improvement from review

jolelievre committed:
Introduce /ProductFormDataProviderTest

jolelievre committed:
Last PR review feedback

jolelievre committed:
Add useless unit test for Progi1984

Commit:
Fix duplicate id on floating button

jolelievre committed:
Merge branch '1.7.7.x' into merge-177-280121

jolelievre committed:
Ignore PHPStan error

jolelievre committed:
Fix navbar conflict and lint

matks committed:
Merge pull request #22921 from jolelievre/product-seo

Product page SEO tab

nesrineabdmouleh committed:
Add an ID for save button

nesrineabdmouleh committed:
Fix an issue

Commit:
Run breakline rule

Progi1984 committed:
Merge pull request #22888 from NeOMakinG/issue22644

Change create order buttons position and wording

nesrineabdmouleh committed:
Fix button name

nesrineabdmouleh committed:
Fix btn name

Commit:
Update stylelint and run fix

Progi1984 committed:
Merge pull request #23037 from nesrineabdmouleh/fixNightly28/01

Fix nightly 28/01/2021

Commit:
Merge pull request #23030 from NeOMakinG/issue23029

Fix duplicate id on floating button

matks committed:
Merge pull request #23031 from jolelievre/merge-177-280121

Merge 177x to develop 2021/01/28

nesrineabdmouleh committed:
Fix default carrier select selector

nesrineabdmouleh committed:
Fix sort by select selector

nesrineabdmouleh committed:
Fix filter logs by employee

nesrineabdmouleh committed:
Fix handling charges input selector

Hlavtox committed:
Update responsive classes

Hlavtox committed:
Update customer.html.twig

Progi1984 committed:
Merge pull request #23006 from Seb33300/patch-1

Fix product search by ISBN

Hlavtox committed:
Update Product.php

Hlavtox committed:
Update ProductController.php

Progi1984 committed:
Merge pull request #23049 from nesrineabdmouleh/fixNightly29/01

Fix nightly 29/01/2021

Progi1984 committed:
Merge pull request #22746 from PierreRambaud/fix/22711

PHP sessions are not correctly configured

Progi1984 committed:
Merge pull request #23034 from NeOMakinG/add-stylelint-breakline

Update stylelint with breakline rule and run fix with it

Progi1984 committed:
Merge pull request #23055 from Hlavtox/patch-16

Add EAN13 and MPN to combinations data

Progi1984 committed:
Merge pull request #23052 from Hlavtox/patch-15

Improve responsivity of customer block in order admin

JevgenijVisockij committed:
Added proper error messages for exceptions during theme upload

JevgenijVisockij committed:
Merge remote-tracking branch 'origin/feature/simplify-theme-form' into feature/simplify-theme-form

JevgenijVisockij commented:
Something went really wrong during rebase and because this PR only had 4 commits it was easier to add them again then to deal with rebase mess, so closes in favor of: 
https://github.com/PrestaShop/PrestaShop/pull/23127
